#! /bin/sh -e
## 10_conf-space-fix.dpatch by Thomas Hood <jdthood@yahoo.co.uk>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Handle spaces in sound card names

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@
diff -u -r1.57 conf.c
--- alsa-lib/src/conf.c	13 Jan 2005 15:13:05 -0000	1.57
+++ alsa-lib/src/conf.c	9 Feb 2005 16:56:04 -0000
@@ -2792,12 +2792,10 @@
 				char *name;
 				if ((err = snd_config_get_ascii(n, &name)) < 0)
 					goto _err;
-				if ((err = snd_user_file(name, &fi[idx].name)) < 0) {
-					SNDERR("\"%s\" is not a word", name);
+				if ((err = snd_user_file(name, &fi[idx].name)) < 0)
+					fi[idx].name = name;
+				else
 					free(name);
-					goto _err;
-				}
-				free(name);
 				idx++;
 				hit = 1;
 			}
